### PR TITLE
Add troubleshooting to environment variables section

### PR DESCRIPTION
**Pull request by blainelewis1**
I recently ran into some issues with using environment variables in a gatsby install which were frustrating so I would like to update the documentation to make it easier to find and fix those. Although I think the first error with the webpack addition of process should probably just be included by default.

## Description

Added troubleshooting documentation for working with environment variables.

### Documentation

It is a documentation based pull request. 

## Related Issues

None that I am aware of.

### CODE REVIEW

**Comment by LekoArts** [comment]
Hi, thanks for the PR!

I think there's a better place to put the first bit of information: https://www.gatsbyjs.com/docs/how-to/local-development/troubleshooting-common-errors/
While we mention it in the migration guide (https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#process-is-not-defined) it's not too discoverable.

I doesn't belong in the environment variables guide as `process.browser` is something different (and it's not needed to make `process.env` working). The other section we haven't heard any reports about this yet so I wouldn't add it (yet) to keep the docs concise.

So TL;DR: We can add the process.browser part to https://www.gatsbyjs.com/docs/how-to/local-development/troubleshooting-common-errors/ using this information: https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#process-is-not-defined

**Comment by blainelewis1** [comment]
Ah okay I see what's happened here. 

I originally added the polyfill because I was getting the same error as the process.browser error (process is not defined), and that's what I thought would help. But what was actually happening is that gatsby doesn't create the process.env variable at all unless there is at least one environment variable defined AND used. And since I "wasn't" using any because of the optional chaining, process.env wasn't being created.  I will rewrite the documentation in environment variables to help people trouble shoot that issue. I can also add the process.browser pieces to troubleshooting-common-issues when I do that too if you would like. 

Thanks!

**Review by mygoldpaw — APPROVED** [review]

**Comment by LekoArts** [comment]
Hi, I'm closing this PR as it got stale. If you still want to put the information in the migration guide, please let me know. Thanks!
